lib/pbench/test/__init__.py

```diff
@@ -8,27 +8,31 @@
 
 def on_disk_config(
     tmp_path_factory, prefix: str, setup_func: Callable[[Path], Path]
-) -> dict:
-    """Base on-disk configuration recorder.  It returns a dictionary with two
-    items, "tmp" and "cfg_dir", two Path objects for the temporary directory
-    in which the configuration directory is created.
+) -> dict[Path, Path]:
```

Comment:
Technically, this should be `Dict[str, Path]` because you're mapping a string key to a `Path` value ...

Reply:
BTW:

```
*: Note: If you are using Python 3.7 or newer, you can use dict as a generic type if you start your module with [from __future__ import annotations](https://www.python.org/dev/peps/pep-0563/), and as of Python 3.9, dict (as well as other standard containers) [supports being used as generic type even without that directive](https://www.python.org/dev/peps/pep-0585/).
```

We still have Python 3.6 platforms (RHEL 8), so until we're "all 9 all the time" we should stick with using the `typing` symbols for collections rather than try to use the raw type as a generic.

Reply:
I am sorry, I really don't see the value of Python type hints.  They serve more to obfuscate code that add clarity, and they just cause additional imports to be added to modules just to reference the types, and don't help during execution.  I'll fix this, but we really need to review how much value they add.

Reply:
I think the *documentation* is useful, both for the developer and for subsequent readers. And we could be running a static analysis tool that actually checks this, which would probably be a good idea.

But you're right that the Python compiler doesn't care, so I won't complain that you didn't use `Dict[str, Path]`... and `dict[str, Path]` is wrong until Python 3.9. I'm not sure what Python we're running in the Jenkins container, and if it's 3.9 I hope this won't cause problems on earlier versions. However, for now I'm going to approve and resolve this conversation.